### ISSUE TITLE
Configure on AIX 6.1 failure OpenSSL 3.0.5

### ISSUE THREAD

**Issue by OneTop4458** [openssl/openssl#19151]
AIX build is failing with OpenSSL 3.0.5

```
[ZEUS:/home/test/workspace/Prebuild/src/openssl-3.0.5 90 ] ./config --prefix=/home/test/openssl/openssl-build --openssldir=/home/test/openssl/openssl-build/ssl  no-asm no-idea no-mdc2 no-rc5 no-dso threads shared

Failure!  build file wasn't produced.
Please read INSTALL.md and associated NOTES-* files.  You may also have to
look over your available compiler tool chain or change your configuration.

Can't use string ("1") as a HASH ref while "strict refs" in use at /home/test/workspace/Prebuild/src/openssl-3.0.5/util/perl/OpenSSL/config.pm line 925.
```

This can be fixed with this  for config.pm:
https://github.com/openssl/openssl/blob/openssl-3.0.5/util/perl/OpenSSL/config.pm#L925
```
return %$result; -> return "\"$result\"";
```

**Comment by OneTop4458** [comment]
Related bugs : https://github.com/openssl/openssl/issues/18119

**Comment by t8m** [comment]
Perhaps @levitte could take a look?

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of the 3.4 dev cycle barring further input
